crowbar_framework/app/models/service_object.rb

```diff
@@ -1543,9 +1561,10 @@ def run_remote_chef_client(node, command, logfile_name)
 
   private
 
-  def wait_for_chef_clients(node)
-    unless RemoteNode.chef_ready?(node, 1200)
-      STDERR.puts "Waiting for already running chef-clients on #{node} failed"
+  def wait_for_chef_clients(node_name)
+    @logger.debug("wait_for_chef_clients: Waiting for already running chef-clients on #{node_name}.")
+    unless RemoteNode.chef_ready?(node_name, 1200)
```

Comment:
The problem is not the logging improvement, the problem is that you're re-using `wait_for_chef_clients` that is really written with the idea that it's going to be called from a `fork`, hence its use of `puts`. And you're using it from inside the rails app instead of from a `fork`.

So the patch itself is not bad; it's just built on something that was not intended for this use.